doc/dev/sharp_bits.rst

```diff
@@ -0,0 +1,1015 @@
+Debugging and sharp bits
+========================
```

Comment:
I was thinking a table of content might be nice to jump to the relevant sections and see at a glance what topics are covered.

Reply:
There is a table of contents automatically on the rhs!

Reply:
I don't see it anywhere in the preview 👀

Reply:
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/e2f186ca-d530-49b8-9fd4-53062506a631)

Reply:
Maybe this a screen size thing?

Reply:
Ah, yes - your window might be slightly too narrow 🤔

Reply:
This is a story I can make for the cloud team to fix